basics/src/main/java/basics/ConvertibleCar.java

```diff
@@ -0,0 +1,35 @@
+package basics;
+
+public class ConvertibleCar extends Car {
+	
+	private boolean TopMotor;
+	
+	public ConvertibleCar() {
+		this.model = null;
+		this.odometer = 0;
+		engine = new SmallEngine();
```

Comment:
This violates Dependency Inversion principle. Basically we can only have cars with LargeEngines. Should pass in IEngine as a parameter instead.